src/containerapp/azext_containerapp/_dapr_utils.py

```diff
@@ -0,0 +1,338 @@
+        component_def = None
+        try:
+            component_def = DaprComponentPreviewClient.show(
+                cmd, resource_group_name, environment_name, component_name
+            )
+        except Exception:  # pylint: disable=broad-except
+            pass
+
+        # Throw an error if the component already exists, and was not created by the init command.
+        # This is to prevent users from accidentally overwriting components that they have created.
+        if component_def and not DaprUtils._is_component_created_by_daprcomponentinit(
+            component_def
+        ):
+            raise ValidationError(
+                f"Dapr component {component_name} already exists and cannot be overwritten."
+                " Please delete the component and try again."
+            )
+
+        # Create the component.
+        logger.debug("Creating Dapr component %s", component_name)
+        component_model = DaprUtils._get_dapr_component_model_from_service(
+            component_type,
+            service_type,
+            service_name,
+            service_id,
+            component_metadata=component_metadata,
+        )
+        try:
+            component_def = DaprComponentPreviewClient.create_or_update(
+                cmd,
+                resource_group_name,
+                environment_name,
+                component_name,
+                component_model,
+            )
+        except Exception as e:
+            raise ValidationError(
+                f"Failed to create Dapr component {component_name}: {e}"
+            ) from e
+
+        if component_def is None:
+            raise ValidationError(
+                f"Failed to create Dapr component {component_name}, component definition is None"
+            )
+
+        logger.debug("Successfully created Dapr component %s", component_name)
+        return component_def
+
+    @staticmethod
+    def _create_service(
+        cmd,
+        service_type: str,
+        service_name: str,
+        resource_group_name: str,
+        environment_name: str,
+    ):
+        """
+        Create a service if it does not exist.
+
+        :param service_type: type of the service to create, e.g. redis
+        :param service_name: name of the service to create, e.g. dapr-redis
+
+        :return: service definition of the service (whether it was created or not)
+        """
+        supported_services = DaprUtils._get_supported_services()
+        if service_type not in supported_services:
+            raise ValidationError(f"Service type {service_type} is not supported.")
+
+        # Look up the service, if it already exists, return it.
+        logger.debug("Looking up service %s of type %s", service_name, service_type)
+        service_def = None
+        try:
+            service_def = ContainerAppPreviewClient.show(
+                cmd, resource_group_name, service_name
+            )
+        except Exception:  # pylint: disable=broad-except
```

Comment:
Please use `handle_non_404_status_code_exception` to avoid ignoring other types of errors.